File: v6/sprefa-extract/tests/fixtures/tsi/probe_graph.go
// The go twin of probe_graph.rs: every form the syntax tier states a row for.
package probe

import "fmt"

type Node[T any, K comparable] struct {
	Value T
	Next  *Node[T, K]
	Tags  []string
	Index map[K]int64
	Base
}

type Base struct {
	ID int64
}

type Shape interface {
	Area(scale float64) float64
	fmt.Stringer
}

type Number interface {
	~int | string
}

func (n *Node[T, K]) Render(width int, pretty bool) (string, error) {
	return "", nil
}

func (n *Node[T, K]) Len() int {
	return 0
}

func Sum[T Number](values []T) T {
	var total T
	return total
}

type Label = string

type Meters int64

const Limit int64 = 10

const Loose = 3

var Flag bool

var Head *Node[int64, string]

func Encode(payload []byte, wide bool) byte {
	return payload[0]
}

File: v6/sprefa-extract/tests/fixtures/tsi/probe_graph.kt
// The kotlin twin of probe_graph.go: every form the syntax tier states a row for.
package probe

import kotlin.collections.List

interface Shape {
    fun area(scale: Double): Double
    val name: String
}

open class Base(val id: Int, label: String) {
    var label: String = label
    fun render(width: Int, pretty: Boolean): String = ""
    fun size(): Int = 0
}

data class Node<T : Base, K>(
    val value: T,
    val tags: List<String>,
    val index: Map<K, Int>,
    val parent: Base?,
) : Base(0, "node"), Shape {
    override val name: String = "node"
    override fun area(scale: Double): Double = 0.0
    fun <R> map(transform: (T) -> R): Node<R, K>? = null
}

sealed class Result {
    data class Ok(val value: Int) : Result()
    data class Err(val message: String, val cause: Throwable?) : Result()
    object Empty : Result()
}

enum class Color { RED, GREEN, BLUE }

object Registry {
    val shapes: MutableList<Shape> = mutableListOf()
    fun register(shape: Shape) { shapes.add(shape) }
}

typealias Label = String
typealias Handler = (Int, String) -> Unit
typealias Lookup<V> = Map<String, V>

fun encode(payload: ByteArray, wide: Boolean = false, vararg rest: Int): ByteArray = payload

fun <T : Comparable<T>> total(values: List<T>): T = values[0]

fun Base.describe(): String = label

fun missing() {}

val limit: Int = 10
val loose = 3
var flag: Boolean = false
var head: Node<Int, String>? = null
